bats/foreign-keys.bats

```diff
@@ -0,0 +1,1236 @@
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+);
+SQL
+    run dolt index ls parent
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "dolt_fk_1(id) HIDDEN" ]] || false
+    run dolt index ls child
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "dolt_fk_1(parent_id) HIDDEN" ]] || false
+    run dolt schema show child
+    [ "$status" -eq "0" ]
+    [[ `echo "$output" | tr -d "\n" | tr -s " "` =~ 'CONSTRAINT `fk_named` FOREIGN KEY (`parent_id`) REFERENCES `parent` (`id`) ON DELETE CASCADE ON UPDATE CASCADE' ]] || false
+}
+
+@test "foreign-keys: CREATE TABLE Single Unnamed FOREIGN KEY Multi-column" {
+    dolt sql <<SQL
+CREATE TABLE parent (
+  id INT PRIMARY KEY,
+  v1 VARCHAR(20),
+  v2 DATETIME
+);
+CREATE TABLE child (
+  id INT PRIMARY KEY,
+  parent_v1 VARCHAR(20),
+  parent_v2 DATETIME,
+  FOREIGN KEY (parent_v1, parent_v2)
+    REFERENCES parent(v1, v2)
+);
+SQL
+    run dolt index ls parent
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "dolt_fk_1(v1, v2) HIDDEN" ]] || false
+    run dolt index ls child
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "dolt_fk_1(parent_v1, parent_v2) HIDDEN" ]] || false
+    run dolt schema show child
+    [ "$status" -eq "0" ]
+    [[ `echo "$output" | tr -d "\n" | tr -s " "` =~ 'CONSTRAINT `fk_child_parent_1` FOREIGN KEY (`parent_v1`,`parent_v2`) REFERENCES `parent` (`v1`,`v2`)' ]] || false
+}
+
+@test "foreign-keys: CREATE TABLE Single Named FOREIGN KEY Multi-column" {
+    dolt sql <<SQL
+CREATE TABLE parent (
+  id INT PRIMARY KEY,
+  v1 VARCHAR(20),
+  v2 DATETIME
+);
+CREATE TABLE child (
+  id INT PRIMARY KEY,
+  parent_v1 VARCHAR(20),
+  parent_v2 DATETIME,
+  CONSTRAINT fk_name FOREIGN KEY (parent_v1, parent_v2)
+    REFERENCES parent(v1, v2)
+);
+SQL
+    run dolt index ls parent
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "dolt_fk_1(v1, v2) HIDDEN" ]] || false
+    run dolt index ls child
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "dolt_fk_1(parent_v1, parent_v2) HIDDEN" ]] || false
+    run dolt schema show child
+    [ "$status" -eq "0" ]
+    [[ `echo "$output" | tr -d "\n" | tr -s " "` =~ 'CONSTRAINT `fk_name` FOREIGN KEY (`parent_v1`,`parent_v2`) REFERENCES `parent` (`v1`,`v2`)' ]] || false
+}
+
+@test "foreign-keys: CREATE TABLE Multiple Unnamed FOREIGN KEY" {
+    dolt sql <<SQL
+CREATE TABLE parent (
+  id INT PRIMARY KEY,
+  v1 VARCHAR(20),
+  v2 DATETIME,
+  v3 TIMESTAMP,
+  v4 BIT(4)
+);
+CREATE TABLE child (
+  id INT PRIMARY KEY,
+  parent_v1 VARCHAR(20),
+  parent_v2 DATETIME,
+  parent_v3 TIMESTAMP,
+  parent_v4 BIT(4),
+  FOREIGN KEY (parent_v1)
+    REFERENCES parent(v1),
+  FOREIGN KEY (parent_v2)
+    REFERENCES parent(v2)
+    ON UPDATE CASCADE,
+  FOREIGN KEY (parent_v3)
+    REFERENCES parent(v3)
+    ON DELETE CASCADE,
+  FOREIGN KEY (parent_v4)
+    REFERENCES parent(v4)
+    ON DELETE CASCADE
+    ON UPDATE CASCADE
+);
+SQL
+    run dolt index ls parent
+    [ "$status" -eq "0" ]
+    [[ "$output" =~ "dolt_fk_1(v1) HIDDEN" ]] || false

```

Comment:
I wouldn't match the fk name in these tests. If the behavior is undefined, don't test for it

Reply:
Could just match the columns, e.g. "(v1) HIDDEN"